### ISSUE TITLE
FieldError cannot retrieves error message from chained path 

### ISSUE THREAD

**Issue by LBrian** [redwoodjs/redwood#2979]
I did a quick search in the `Issues` and `Pull requests` and seems nothing similar is reported yet, but feel free to close it if it is redundant.

This is not a blocking issue or must-have at all but more like a better-to-have improvement as I did run into the illusion and believe this could be prevented.

Currently, `FieldError` retrieves error message using `errors[props.name]` which doesn't support chained path, e.g. `address.street`
```jsx
const FieldError = (props: FieldErrorProps) => {
  const { errors } = useFormContext()
  const validationError = errors[props.name]  // chained path is not supported
  const errorMessage =
    validationError &&
    (validationError.message ||
      `${props.name} ${
        DEFAULT_MESSAGES[validationError.type as keyof typeof DEFAULT_MESSAGES]
      }`)

  return validationError ? <span {...props}>{errorMessage}</span> : null
}
```

since `react-hook-form` support chained path, so this will be enhanced if form structure has nested data structure as below, `address.name` and `address.street` data can be set/get follows the data structure but `FeildError` cannot retrieve error message coz `errors['address.name']` won't work
```jsx
  <Form onSubmit={onSubmit}>
      <TextField
          name="firstName"
          disabled={loading}
          placeholder="e.g. John"
          defaultValue={firstName}
          className="input input-sm input-bordered w-full"
          errorClassName="input input-bordered input-error"
        />
        <FieldError name="firstName" className="mt-1 label-text-alt text-error" />
       <TextField
          name="address.name"
          disabled={loading}
          placeholder="e.g. My house"
          defaultValue={address?.name}
          className="input input-sm input-bordered w-full"
          errorClassName="input input-bordered input-error"
        />
        <FieldError name="address.name" className="mt-1 label-text-alt text-error" />
     <TextField
          name="address.street"
          disabled={loading}
          placeholder="e.g. 120/20 George St."
          defaultValue={address?.street}
          className="input input-sm input-bordered w-full"
          errorClassName="input input-bordered input-error"
        />
        <FieldError name="street" className="mt-1 label-text-alt text-error" />
 </Form>
```
luckily, `react-hook-form` exports `get` utility does the same thing as `lodash` `_.get` and should be just one-liner change for this improvement
https://github.com/react-hook-form/react-hook-form/blob/master/src/utils/get.ts

I can create a PR for this once we confirm it, so feel free to assign to me. cc @Tobbe @peterp

**Comment by thedavidprice** [comment]
Thanks @LBrian Looks like this could be done in conjunction with the upgrade to RHF v7 via this PR https://github.com/redwoodjs/redwood/pull/2270 (which is long overdue for some TLC). Thoughts about this @jtoar?

Also, @LBrian if you are looking to help we could use extra hands on #2270

**Comment by Tobbe** [comment]
@thedavidprice has already given an excellent answer, but since you tagged me I wanted to answer too.

#2270 is the bigger, more important, PR. But if you feel you don't have time to contribute to that one, please do feel free to make a separate PR for this issue. We always appreciate any kind of improvement, no matter how big or small! :)

**Comment by LBrian** [comment]
This is an easy fix and I am happy to help on #2270 but as Tobbe said RHF v7 upgrade is quite big and my time is fragile as I have few ongoing side projects, so I will create a separated PR for this fix then jump to #2270 using my fragile time.

**Comment by thedavidprice** [comment]
Thank you @LBrian! 

And appreciate you helping to connect the dots and suggest options @Tobbe!
